ihe-interop/src/test/java/org/motechproject/ihe/interop/handler/helper/IHEActionHelperTest.java

```diff
@@ -80,10 +83,10 @@ public void shouldCallSendTemplateMethod() throws IOException, SAXException, Par
             byteObjects[i++] = b;
         }
 
-
         when(iheTemplateDataService.findByName("sampleTemplate3")).thenReturn(cdaTemplate);
         when(hl7RecipientsService.getRecipientbyName("sampleRecipient3")).thenReturn(hl7Recipient);
         when((Byte[]) iheTemplateDataService.getDetachedField(cdaTemplate, "templateData")).thenReturn(byteObjects);
+        verify(iheTemplateService).sendTemplateToRecipientUrl(new String(bytes, StandardCharsets.UTF_8), new PostMethod(hl7Recipient.getRecipientUrl()));
```

Comment:
Why is this verify() before method call?  It should throw an exception here. I checked this and there is some problem with hl7Recipient.getRecipientUrl() - without this clause the "Wanted but not invoked" exception appears.